web/src/settings_components.ts

```diff
@@ -424,6 +437,26 @@ export function read_field_data_from_form(
     return undefined;
 }
 
+function get_field_data_input_value($input_elem: JQuery): string | undefined {
+    const $profile_field_form = $input_elem.parent();
+    const profile_field_id = Number.parseInt(
+        $($profile_field_form).attr("data-profile-field-id")!,
+        10,
+    );
+    const field = realm.custom_profile_fields.find((field) => field.id === profile_field_id)!;
```

Comment:
We could've introduced `field` param in `get_input_element_value`, but I've kept it clean since it didn't have any similar kind of data input. The performance difference due to this should be negligible imo. LMK if we might wanna do this the alternate way

Reply:
Yeah performance is definitely negligible -- it's not a matter of opinion that scanning a list of a few dozen elements once in a given UI interaction can't add up.